generated-package/K8sObject.pkl

```diff
@@ -131,6 +131,10 @@ const function hasNonNullPortNames(l: Listing): Boolean = (l.length > 1).implies
 const function exactlyOneSet(a: Any, b: Any, c: Any) =
   (a != null).xor(b != null).xor(c != null) && !(a != null && b != null && c != null)
 
+/// Tells if exactly one of the four arguments is non-null.
+const function exactlyOneSet4(a: Any, b: Any, c: Any, d: Any) =
+  (a != null).xor(b != null).xor(c != null).xor(d != null) && !(a != null && b != null && c != null && d != null)
```

Comment:
I think this will return true if 3/4 are true

Reply:
```
pkl1> a = null; b = "x" ; c = 2 ; d = true
pkl2> (a != null).xor(b != null).xor(c != null).xor(d != null) && !(a != null && b != null && c != null && d != null)
true
```

Reply:
Good catch, updated.

I don't know if there's a simpler way to write this expression asides from my updated version.

Reply:
You could do a summation of `a != null ? 1 : 0 + ... == 1` but I'm not sure if that comes out simpler or more readable